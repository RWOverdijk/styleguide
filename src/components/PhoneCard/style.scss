/** @define PhoneCard */

@import "../../css/variables";
@import "../../css/mixins";

.Van-PhoneCard {
  width: 325px;
  min-height: 75px;
  background-color: rgba(255, 255, 255, 0.6);
  padding: 15px 10px;
  text-align: left;
  display: inline-block;
  align-items: flex-start;

  &--withBorder {
    border: 2px solid $color-grey-bright;
  }

  &-number {
    @include color-fill($color-green-primary);

    display: inline-flex;
    font-weight: $font-weight-semibold;
    font-size: 17px;
    line-height: 25px;
  }

  /* postcss-bem-linter: ignore */
  .Van-Icon {
    @include icon-size(23px);

    margin-right: 8px;
    margin-top: 0;
  }

  &-description {
    display: block;
    color: $color-primary-bright;
    font-size: 13px;
    line-height: 18px;
  }
}
