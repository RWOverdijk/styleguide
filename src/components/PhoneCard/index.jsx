import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './style.scss';

class PhoneCard extends PureComponent {
  getClass(className) {
    let classes = ['Van-PhoneCard'];

    if (this.props.className) {
      classes.push(this.props.className);
    }

    classes.push(className);

    return classes.join(' ').trim();
  }

  render() {
    const {
      className,
      ...props
    } = this.props;

    return (
      <div className={this.getClass()} {...props}>
        <span className="Van-PhoneCard-number">
          <i className="fa fa-phone" />
          {this.props.number}
        </span>
        <span className="Van-PhoneCard-description">{this.props.description}</span>
      </div>
    );
  }
}

PhoneCard.propTypes = {
  className: PropTypes.string,
  number: PropTypes.string,
  description: PropTypes.string
};

export default PhoneCard;
