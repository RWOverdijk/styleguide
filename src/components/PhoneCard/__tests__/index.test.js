import React from 'react';
import renderer from 'react-test-renderer';
import PhoneCard from '../';

describe("<PhoneCard>", () => {
  it("renders PhoneCard", () => {
    const tree = renderer.create(
      <PhoneCard
        number="1234567890"
        description="PhoneCard description"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders PhoneCard with border", () => {
    const tree = renderer.create(
      <PhoneCard
        withBorder
        number="1234567890"
        description="PhoneCard description"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <PhoneCard
        number="1234567890"
        description="PhoneCard description"
        className="my-class"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <PhoneCard
        number="1234567890"
        description="PhoneCard description"
        data-id="0"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
