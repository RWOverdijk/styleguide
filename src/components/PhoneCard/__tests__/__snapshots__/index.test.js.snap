// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneCard> renders PhoneCard 1`] = `
<div
  className="Van-PhoneCard"
  description="PhoneCard description"
  number="1234567890"
>
  <span
    className="Van-PhoneCard-number"
  >
    <span
      className="Van-Icon"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
      source="phone"
    />
    1234567890
  </span>
  <span
    className="Van-PhoneCard-description"
  >
    PhoneCard description
  </span>
</div>
`;

exports[`<PhoneCard> renders PhoneCard with border 1`] = `
<div
  className="Van-PhoneCard Van-PhoneCard--withBorder"
  description="PhoneCard description"
  number="1234567890"
>
  <span
    className="Van-PhoneCard-number"
  >
    <span
      className="Van-Icon"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
      source="phone"
    />
    1234567890
  </span>
  <span
    className="Van-PhoneCard-description"
  >
    PhoneCard description
  </span>
</div>
`;

exports[`<PhoneCard> renders with specific attributes 1`] = `
<div
  className="Van-PhoneCard"
  data-id="0"
  description="PhoneCard description"
  number="1234567890"
>
  <span
    className="Van-PhoneCard-number"
  >
    <span
      className="Van-Icon"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
      source="phone"
    />
    1234567890
  </span>
  <span
    className="Van-PhoneCard-description"
  >
    PhoneCard description
  </span>
</div>
`;

exports[`<PhoneCard> renders with specific className 1`] = `
<div
  className="Van-PhoneCard my-class"
  description="PhoneCard description"
  number="1234567890"
>
  <span
    className="Van-PhoneCard-number"
  >
    <span
      className="Van-Icon"
      dangerouslySetInnerHTML={
        Object {
          "__html": "test-file-stub",
        }
      }
      source="phone"
    />
    1234567890
  </span>
  <span
    className="Van-PhoneCard-description"
  >
    PhoneCard description
  </span>
</div>
`;
